homeassistant/components/withings/config_flow.py

```diff
@@ -1,192 +1,64 @@
-        if user_input.get(const.CODE):
-            self.data = user_input
-            return self.async_external_step_done(next_step_id="finish")
-
-        profile = user_input.get(const.PROFILE)
-
-        auth_client = self.get_auth_client(profile)
-
-        url = auth_client.get_authorize_url()
-
-        return self.async_external_step(step_id="auth", url=url)
-
-    async def async_step_finish(self, user_input=None):
-        """Received code for authentication."""
-        data = user_input or self.data or {}
-
-        _LOGGER.debug(
-            "Should close all flows below %s",
-            self.hass.config_entries.flow.async_progress(),
-        )
-
-        profile = data[const.PROFILE]
-        code = data[const.CODE]
-
-        return await self._async_create_session(profile, code)
-
-    async def _async_create_session(self, profile, code):
-        """Create withings session and entries."""
-        auth_client = self.get_auth_client(profile)
-
-        _LOGGER.debug("Requesting credentials with code: %s.", code)
-        credentials = auth_client.get_credentials(code)
-
-        return self.async_create_entry(
-            title=profile,
-            data={const.PROFILE: profile, const.CREDENTIALS: credentials.__dict__},
-        )
-
-
-class WithingsAuthCallbackView(HomeAssistantView):
-    """Withings Authorization Callback View."""
-
-    requires_auth = False
-    url = const.AUTH_CALLBACK_PATH
-    name = const.AUTH_CALLBACK_NAME
-
-    def __init__(self):
-        """Constructor."""
-
-    async def get(self, request):
-        """Receive authorization code."""
-        hass = request.app["hass"]
-
-        code = request.query.get("code")
-        profile = request.query.get("profile")
-        flow_id = request.query.get("flow_id")
-
-        if not flow_id:
-            return aiohttp.web_response.Response(
-                status=400, text="'flow_id' argument not provided in url."
-            )
-
-        if not profile:
-            return aiohttp.web_response.Response(
-                status=400, text="'profile' argument not provided in url."
-            )
-
-        if not code:
-            return aiohttp.web_response.Response(
-                status=400, text="'code' argument not provided in url."
-            )
-
-        try:
-            await hass.config_entries.flow.async_configure(
-                flow_id, {const.PROFILE: profile, const.CODE: code}
-            )
-
-            return aiohttp.web_response.Response(
-                status=200,
-                headers={"content-type": "text/html"},
-                text="<script>window.close()</script>",
-            )
+    async def async_step_finish(self, data: dict) -> dict:
+        """Finish the flow."""
+        self._current_data = None
 
-        except data_entry_flow.UnknownFlow:
-            return aiohttp.web_response.Response(status=400, text="Unknown flow")
+        return await super().async_oauth_create_entry(data)
```

Comment:
Are you able to figure out the name of the created profile and use that as title for the entry ?